File: crates/rsync-core/src/models/progress.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use super::job::JobStatus;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ProgressUpdate {
    pub invocation_id: Uuid,
    pub bytes_transferred: u64,
    pub percentage: f64,
    pub transfer_rate: String,
    pub elapsed: String,
    pub files_transferred: u64,
    pub files_remaining: u64,
    pub files_total: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct LogLine {
    pub invocation_id: Uuid,
    pub timestamp: DateTime<Utc>,
    pub line: String,
    pub is_stderr: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct JobStatusEvent {
    pub job_id: Uuid,
    pub invocation_id: Uuid,
    pub status: JobStatus,
    pub exit_code: Option<i32>,
    pub error_message: Option<String>,
}

File: src/types/execution/progress.ts
import type { JobStatus } from "../job";

export interface ProgressUpdate {
  invocation_id: string;
  bytes_transferred: number;
  percentage: number;
  transfer_rate: string;
  elapsed: string;
  files_transferred: number;
  files_remaining: number;
  files_total: number;
}

export interface LogLine {
  invocation_id: string;
  timestamp: string;
  line: string;
  is_stderr: boolean;
}

export interface JobStatusEvent {
  job_id: string;
  invocation_id: string;
  status: JobStatus;
  exit_code: number | null;
  error_message: string | null;
}
